WooCommerce/Classes/ViewModels/MainTabViewModel.swift

```diff
@@ -16,11 +17,28 @@ extension NSNotification.Name {
 }
 
 final class MainTabViewModel {
+
+    private let storesManager: StoresManager
+    private let featureFlagService: FeatureFlagService
+
+    init(storesManager: StoresManager = ServiceLocator.stores,
+         featureFlagService: FeatureFlagService = ServiceLocator.featureFlagService) {
+        self.storesManager = storesManager
+        self.featureFlagService = featureFlagService
+    }
+
     /// Callback to be executed when this view model receives new data
     /// passing the string to be presented in the badge as a parameter
     ///
     var onBadgeReload: ((String?) -> Void)?
 
+    /// Must be called during `MainTabBarController.viewDidAppear`. This will try and save the
+    /// app installation date.
+    ///
+    func onViewDidAppear() {
+        saveInstallationDateIfNecessary()
+    }
```

Comment:
Since `saveInstallationDateIfNecessary()` is called only on `MainTabBarController.viewDidAppear`, maybe it's better to move the note on top of `saveInstallationDateIfNecessary` function it self, making it public, and remove the `onViewDidAppear()` function? In that way maybe it's more clear what the MainTabBarController is doing, saving the installation date.

Reply:
Make totally sense! Thank you @shiki for explaining it 👍